sdk/storage/azure-storage-file-share/azure/storage/fileshare/_serialize.py

```diff
@@ -84,35 +88,35 @@ def get_source_conditions(kwargs):
     )
 
 
-def get_access_conditions(lease):
-    # type: (ShareLeaseClient or str) -> LeaseAccessConditions or None
+def get_access_conditions(lease: Optional[Union["ShareLeaseClient", str]]) -> Optional[LeaseAccessConditions]:
     try:
-        lease_id = lease.id # type: ignore
+        lease_id = lease.id  # type: ignore
     except AttributeError:
-        lease_id = lease # type: ignore
+        lease_id = lease  # type: ignore
```

Comment:
For the three lease related things in here, how about this?

```py
if isinstance(lease, ShareLeaseClient):
    lease_id = lease.id
else:
    lease_id = lease
```

I _think_ MyPy should be happy with that and not need any ignores.

Reply:
Looks like there's an additional `None` check, so will add that in and let me know how you feel about this style.